tests/integration-tests/tests/iam/test_iam_image.py

```diff
@@ -106,9 +119,11 @@ def _check_roles(
             # Check the role is attached to the Lambda function
             lambda_function = lambda_client.get_function(FunctionName=resource["PhysicalResourceId"])["Configuration"]
             assert_that(lambda_function["Role"]).is_equal_to(lambda_cleanup_role)
+            logging.info("Lambda function role confirmed")
         if resource_type == "AWS::ImageBuilder::Image":
```

Comment:
We can move this entire logic into the `_get_resources_with_image_resource` renaming it to `_check_instance_profile_arn`.

I mean we're doing the `_get_resources_with_image_resource` to retrieve the image arn, then we're using in the check outside, it's better to put together the logic to wait for it and to assert for the policy, since the lambda assertion is a totally separate check.

Reply:
doing it this way will ensure the lambda function resource is also there.  It is possible that neither exists.  I was actually considering also checking for the lambda resource, but the image resource will take longer to start creation

Reply:
We are also retrieving the resources at this step, if I move the wait condition to the if block for the image resource, I will need to get the resources in two separate places

Reply:
ok let's keep it as is! Thanks for the clarification.